app.js

```diff
@@ -0,0 +1,126 @@
+const express = require("express");
+const cors = require("cors")
+const bodyParser = require("body-parser");
+const app = express();
+const port = 4000;
+const School = require("./School");
+const Class = require('./Class');
+const Student = require('./Student')
+
+
+
+app.use(cors());
+app.use(bodyParser.urlencoded({extended:false}));
+app.use(bodyParser.json())
+
+let mySchool = new School();
+
+const checkClassName = (request,response,next)=>{
+if(!request.body.name||!request.body.teacher){
+    response.json({
+        error:"not valid data"
+    })
+}else{
+    next()
+}
+}
+const addClass =(request,response)=>{
+    let newClass = mySchool.addClass(request.body.name,request.body.teacher)
+    if(newClass=== newClass){
+        response.json({
+            error:"class exist",
+            time: new Date()
+        })
+    }else{
+        response.json({
+            class: newClass,
+            message: "created new class",
+            time: new Date()
+        })
+    }
+
+}
+const ifRepeated =(request,response,next)=>{
+let className = request.body.name
+if(!mySchool.classes[className]){
+    response.json({
+        error:"class exists",
+        time: new Date()
+    })
+}
+next()
+
+}
+const enrollStudent=(request,response)=>{
+    let name = request.body.name;
+    let age = request.body.age;
+    let city = request.body.city;
+    let grade = request.body.grade;
+    let className = request.params.name;
+    let newStudent = new Student(name, age, city, grade, className)
+    let enrolledStudent = mySchool.enrollStudent(request.params['className'], newStudent);
+    if(!enrolledStudent){
+        response.json({
+            error: "student does not exist",
+            time: new Date()
+        })
+    }else {
+        response.json({
+            student : enrolledStudent,
+            class: className,
+            message: 'Enrolled ',
+            time: new Date()
+        })
+    }
+}
+const classExist=(request,response,next)=>{
+    let className = (request.params.name)
+    if(!mySchool.classes[className]){
+        response.json({
+
```

Comment:
You can include a message saying "You are trying to enroll the student in a class that doesn't exist in our system. Pick a different class" (smth along those lines)